### PR TITLE
feat(kit): `InputSlider` use new `Slider` inside

### PR DESCRIPTION
**Pull request by nsbarsukov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [X] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: projects/kit/abstract/input-slider.ts [hunk 1]
```diff
@@ -145,6 +172,30 @@ export abstract class AbstractTuiInputSlider<T>
...
+        if (this.segments !== 1) {
+            return '';
+        }
+
+        if (segment === 0) {
+            return `${texts[0]} `;
+        }
+
+        return `${texts[1]} `;
+    }
```
Comment (splincode) [inline projects/kit/abstract/input-slider.ts:186]:
```ts
return segment === 0 ? `${texts[0]}`: `${texts[1]}`;
```

File: projects/kit/components/input-slider/input-slider.style.less [hunk 2]
```diff
@@ -104,18 +104,23 @@
...
     margin: -0.5625rem 0 0;
-    color: transparent;
+}
 
-    :host._disabled &,
-    :host._readonly & {
-        pointer-events: none;
-    }
+// Dont change CSS-selector (we require the such specificity to override default styles without !important)
+input[type='range'] {
```
Comment (splincode) [inline projects/kit/components/input-slider/input-slider.style.less:117]: won't that override styles elsewhere? can this break the layout for someone else? or is it encapsulated by angular?

File: projects/kit/components/input-slider/input-slider.template.html [hunk 3]
```diff
@@ -2,7 +2,7 @@
     [appearance]="appearance"
     [readOnly]="readOnly"
     [disabled]="disabled"
-    [focused]="computedFocused"
+    [focused]="computedFocused || sliderFocused"
```
Comment (splincode) [inline projects/kit/components/input-slider/input-slider.template.html:5]: why the `sliderFocused` getter can't it be moved to the `computedFocused` getter?

File: projects/kit/components/input-slider/input-slider.template.html [hunk 4]
```diff
@@ -65,22 +65,41 @@
...
+
+    <!--TODO delete segments labels in v3.0-->
+    <div *ngIf="segmented && !isNew" class="t-segments-labels">
+        <span
+            *tuiRepeatTimes="let segmentIndex of segments + 1"
+            automation-id="tui-slider__segment"
+        >
+            <span *ngIf="fromToTexts$ | async as fromToText">
+                {{ getSegmentPrefix(segmentIndex, fromToText) }} {{
+                getSegmentLabel(segmentIndex) | tuiFormatNumber }}
```
Comment (splincode) [inline projects/kit/components/input-slider/input-slider.template.html:97]: `getSegmentLabel` is pure function? will there be performance problem here?

Comment (splincode) [reply projects/kit/components/input-slider/input-slider.template.html:97]: why not 

```
{{ fromToText | segmentPrefix: segmentIndex }} {{ segmentIndex | segmentLabel | tuiFormatNumber }}


```

File: projects/kit/utils/math/slider-key-steps.utils.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    valuePercentage: number,
+    [upperStepPercent, upperStepValue]: [number, number],
+    [lowerStepPercent, lowerStepValue]: [number, number],
+): number {
+    const ratio =
+        (valuePercentage - lowerStepPercent) / (upperStepPercent - lowerStepPercent);
+    const controlValue = (upperStepValue - lowerStepValue) * ratio + lowerStepValue;
+
+    return round(controlValue, TUI_FLOATING_PRECISION);
+}
```
Comment (splincode) [inline projects/kit/utils/math/slider-key-steps.utils.ts:37]: can we write a simple unit test for these functions? so then we can be refactored if needed

File: projects/kit/components/input-slider/input-slider.style.less (hunk 2 above)
Comment (nsbarsukov) [reply projects/kit/components/input-slider/input-slider.style.less:117]: > won't that override styles elsewhere?

Component has `ViewEncapsulation.Emulated`, it should not

File: projects/demo/src/modules/components/input-slider/examples/3/index.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    encapsulation,
+})
+export class TuiInputSliderExample3 {
+    readonly control = new FormControl(10_000);
+    readonly max = 1_000_000;
+    readonly min = 0;
+    readonly totalSteps = 100;
+    readonly ticksLabels = ['0', '10K', '100K', '500k', '1000K'];
+
+    readonly keySteps: TuiKeySteps = [
```
Comment (splincode) [inline projects/demo/src/modules/components/input-slider/examples/3/index.ts:21]: what do you think about supporting one more data structure?

```
readonly keySteps: TuiKeyStepsStruct = [
  { percent: 25, value: 10_000 },
  { percent: 50, value: 100_000 },
  { percent: 75, value: 500_000 }
]
```

Comment (splincode) [reply projects/demo/src/modules/components/input-slider/examples/3/index.ts:21]: purely a suggestion, as a feature in the future

File: projects/kit/utils/math/slider-key-steps.utils.ts (hunk 5 above)
Comment (nsbarsukov) [reply projects/kit/utils/math/slider-key-steps.utils.ts:37]: all math calculation is already tested [here](https://github.com/Tinkoff/taiga-ui/blob/main/projects/kit/components/slider/test/slider-key-steps.spec.ts).

Should we duplicate checks for this functions ?

Comment (splincode) [reply projects/kit/utils/math/slider-key-steps.utils.ts:37]: oh, sorry, no

**Review by splincode — APPROVED** [review]

File: projects/kit/components/input-slider/input-slider.template.html (hunk 4 above)
Comment (nsbarsukov) [reply projects/kit/components/input-slider/input-slider.template.html:97]: It is just copy-paste from old code))

But I agree, pipes is suitable for this.
I'll make it

Comment (nsbarsukov) [reply projects/kit/components/input-slider/input-slider.template.html:97]: and this help to optimize imports... amazing!
thanks!

Comment (nsbarsukov) [reply projects/kit/components/input-slider/input-slider.template.html]: I have made pipe `tuiSliderTickLabel` and marked `getSegmentPrefix` with `@tuiPure` decorator

File: projects/kit/components/input-slider/input-slider.template.html (hunk 3 above)
Comment (nsbarsukov) [reply projects/kit/components/input-slider/input-slider.template.html:5]: oh no, sorry, my last message is not true.

It doesnt update value while drag by another reason.
I'll add a commit with changes

File: projects/kit/abstract/input-slider.ts [hunk 7]
```diff
@@ -136,6 +165,10 @@ export abstract class AbstractTuiInputSlider<T>
         return this.length / this.computedSteps;
     }
 
+    get computedKeySteps(): TuiKeySteps {
+        return [[0, this.min], ...(this.keySteps || []), [100, this.max]];
```
Comment (splincode) [inline projects/kit/abstract/input-slider.ts:169]: I'm not sure about this getter, because for each call to change detection a new object will be created

Comment (splincode) [reply projects/kit/abstract/input-slider.ts:169]: maybe it's better

```ts
get computedKeySteps(): TuiKeySteps {
   return this.computePureKeySteps(this.keySteps, this.min, this.max);
}

@tuiPure
private computePureKeySteps(keySteps, min, max) {
 //...
}
```

File: projects/kit/components/input-slider/input-slider.template.html (hunk 3 above)
Comment (nsbarsukov) [reply projects/kit/components/input-slider/input-slider.template.html]: @splincode I have changed it now as you proposed.

Now `InputSlider` should work in the same way as it was.

File: projects/kit/abstract/input-slider.ts (hunk 7 above)
Comment (nsbarsukov) [reply projects/kit/abstract/input-slider.ts]: ✅

**Comment by splincode** [comment]
Aw, you have some problems, please rebase directly / or squash commits

image

**Comment by splincode** [comment]
also, you can change commit name to `feat(kit): InputSlider use new Slider inside`

**Review by waterplea — APPROVED** [review]

**Pull request merged**